### PR TITLE
Allow passing external solvers as list

### PR DESCRIPTION
**Pull request by vkgnosis**
Fixes https://github.com/cowprotocol/services/issues/150

### Test Plan

CI and manually tested that the argument works

### CODE REVIEW

File: crates/solver/src/main.rs [hunk 1]
```diff
@@ -313,39 +316,6 @@ enum TransactionStrategyArg {
     DryRun,
 }
 
-#[derive(Debug)]
-enum SolverAccountArg {
```
Comment (vkgnosis) [inline crates/solver/src/main.rs:317]: This moved into crates/solver/src/solver.rs

File: crates/solver/src/solver.rs [hunk 2]
```diff
@@ -146,6 +147,62 @@ pub enum SolverType {
     BalancerSor,
 }
 
+#[derive(Debug)]
+pub enum SolverAccountArg {
```
Comment (vkgnosis) [inline crates/solver/src/solver.rs:151]: Moved from solver/src/main.rs

File: crates/solver/src/solver.rs [hunk 3]
```diff
@@ -146,6 +147,62 @@ pub enum SolverType {
...
+                            .context(anyhow!("could not parse as address: {}", addr_err))
+                            .context("invalid solver account, it is neither a private key or an Ethereum address")
+                    },
+                )?))
+            })
+    }
+}
+
+#[derive(Debug)]
+pub struct ExternalSolverArg {
```
Comment (vkgnosis) [inline crates/solver/src/solver.rs:184]: New

File: crates/solver/src/solver.rs [hunk 4]
```diff
@@ -172,12 +229,13 @@ pub fn create(
     quasimodo_uses_internal_buffers: bool,
     mip_uses_internal_buffers: bool,
     one_inch_url: Url,
+    external_solvers: Vec<ExternalSolverArg>,
 ) -> Result<Solvers> {
     // Tiny helper function to help out with type inference. Otherwise, all
     // `Box::new(...)` expressions would have to be cast `as Box<dyn Solver>`.
     #[allow(clippy::unnecessary_wraps)]
```
Comment (MartinquaXD) [inline crates/solver/src/solver.rs:236]: nit: This line is no longer necessary.

**Review by sunce86 — APPROVED** [review]

**Review by MartinquaXD — APPROVED** [review]
LGTM.

**Pull request merged**